src/databricks/labs/ucx/contexts/application.py

```diff
@@ -272,6 +277,23 @@ def table_ownership(self) -> TableOwnership:
             self.workspace_path_ownership,
         )
 
+    @cached_property
+    def default_securable_ownership(self) -> DefaultSecurableOwnership:
+        # validate that the default_owner_group is set and is a valid group (the current user is a member)
+        owner_group = None
+        if self.config.default_owner_group:
+            if not self.group_manager.validate_owner_group(self.config.default_owner_group):
+                logger.warning("Default owner group is not valid, falling back to administrator ownership.")
+            else:
+                owner_group = self.config.default_owner_group
+        # Returns a static table ownership resolver
+        return DefaultSecurableOwnership(
+            self.administrator_locator,
+            self.tables_crawler,
+            owner_group,
+            lambda: self.workspace_client.current_user.me().user_name,
+        )
```

Comment:
this is too much logic for a factory class. reduce this method to 
```suggestion
    def default_securable_ownership(self) -> DefaultSecurableOwnership:
        return DefaultSecurableOwnership(
            self.administrator_locator,
            self.tables_crawler,
            self.group_manager,
            self.config.default_owner_group,
            self.workspace_client,
        )
```
and perform validation in `DefaultSecurableOwnership#_application_principal`

Reply:
Fixed